lib/index.js

```diff
@@ -18,9 +18,28 @@ KeenLibrary.prototype.query = function(a, b = undefined, options = {}){
   let analysisType = a;
   let queryParams = b;
 
+  const isAnalysisTypeAnObject = typeof analysisType === 'object';
+  const savedRegex = /saved/;
+  const isSavedAnalysisType = (isAnalysisTypeAnObject && savedRegex.test(analysisType.analysis_type))
+    || savedRegex.test(analysisType);
+  
+    // validate params
+  if (!isSavedAnalysisType) {
+    if (!this.config.masterKey || !this.config.readKey) throw new Error('Please provide valid API key');
```

Comment:
move this code outside of `if (!isSavedAnalysisType) {` , because we need API auth keys even for saved queries

Reply:
&& instead of || 
`(!this.config.masterKey && !this.config.readKey)` 
- you can run queries using just one of these keys